mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractMessageIdManagerStorageTest.java

```diff
@@ -0,0 +1,280 @@
+    }
+
+    @Test
+    public void deleteMessageShouldNotFailWhenMessageDoesNotExist() throws Exception {
+        MessageId messageId = testingData.createNotUsedMessageId();
+
+        messageIdManager.delete(messageId, ImmutableList.of(mailbox1.getMailboxId()), session);
+    }
+
+    @Test
+    public void setMailboxesShouldNotFailWhenMessageDoesNotExist() throws Exception {
+        MessageId messageId = testingData.createNotUsedMessageId();
+
+        messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox1.getMailboxId()), session);
+    }
+
+    @Test
+    public void getMessagesShouldReturnStoredResults() throws Exception {
+        MessageId messageId = testingData.persist(mailbox1.getMailboxId(), FLAGS);
+
+        assertThat(messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, session))
+            .hasSize(1);
+    }
+
+    @Test
+    public void setInMailboxesShouldSetMessageInBothMailboxes() throws Exception {
+        MessageId messageId = testingData.persist(mailbox1.getMailboxId(), FLAGS);
+
+        messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox2.getMailboxId()), session);
+
+        assertThat(messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, session))
+            .hasSize(2);
+    }
+
+    @Test
+    public void setInMailboxesShouldNotDuplicateMessageIfSameMailbox() throws Exception {
+        MessageId messageId = testingData.persist(mailbox1.getMailboxId(), FLAGS);
+
+        messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox1.getMailboxId()), session);
+
+        assertThat(messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, session))
+            .hasSize(1);
+    }
+
+    @Test
+    public void setInMailboxesShouldSetHighestUidInNewMailbox() throws Exception {
+        MessageId messageId1 = testingData.persist(mailbox1.getMailboxId(), FLAGS);
+        MessageId messageId2 = testingData.persist(mailbox2.getMailboxId(), FLAGS);
+
+        messageIdManager.setInMailboxes(messageId1, ImmutableList.of(mailbox2.getMailboxId()), session);
+
+        MessageUid uidMessage1Mailbox1 = messageIdManager.getMessages(ImmutableList.of(messageId1), FetchGroupImpl.MINIMAL, session)
+            .get(0)
+            .getUid();
+        MessageUid uidMessage2Mailbox1 = FluentIterable
+            .from(messageIdManager.getMessages(ImmutableList.of(messageId2), FetchGroupImpl.MINIMAL, session))
+            .filter(new Predicate<MessageResult>() {
+                @Override
+                public boolean apply(MessageResult input) {
+                    return input.getMailboxId().equals(mailbox1.getMailboxId());
+                }
+            })
+            .toList()
+            .get(0)
+            .getUid();
+
+        assertThat(uidMessage2Mailbox1).isGreaterThan(uidMessage1Mailbox1);
+    }
+
+    @Test
+    public void setInMailboxesShouldSetHighestModSeqInNewMailbox() throws Exception {
```

Comment:
I'm not sure to understand what you test in these two tests setInMailboxesShouldSetHighestUidInNewMailbox and setInMailboxesShouldSetHighestModSeqInNewMailbox. You are using setInMailboxes, but you never check uidMessageXMailbox2. And you already have a setInMailboxesShouldNotChangeUidAndModSeqInOriginalMailbox.